src/sentry/pipeline/base.py

```diff
@@ -94,7 +97,7 @@ def __init__(
         self,
         request: Request,
         provider_key: str,
-        organization: Organization | None = None,
+        organization: ApiOrganization | None = None,
```

Comment:
Will need to see what is the blast radius on this change and whether it makes sense to try to convert all `Pipeline` subclasses at once. It doesn't seem to break a lot immediately, but that's probably only because the subclasses lack Mypy coverage.

As an alternative, it probably would work to type it as `Organization | ApiOrganization | None` temporarily.